src/Microsoft.DotNet.Build.Tasks.Feed/BlobFeedAction.cs

```diff
@@ -109,83 +115,47 @@ public BlobFeedAction(string expectedFeedUrl, string accountKey, string indexDir
             return !Log.HasLoggedErrors;
         }
 
-        private async Task UploadIndexJson(SemaphoreSlim clientThrottle, bool allowOverwrite, string packageDirectory, List<string> listAzureBlobs)
+        private LocalSettings GetSettings()
         {
-            listAzureBlobs.Remove(listAzureBlobs.Find(x => x.Contains("index.json")));
-            List<string> updatedVersions = new List<string>();
-            foreach (var version in listAzureBlobs)
+            
+
+            SleetSettings sleetSettings = new SleetSettings()
             {
-                string versionToCheck = version.Substring(packageDirectory.Length + 1).Split('/')[0];
-                NuGetVersion nugetVersion = null;
-                if (NuGetVersion.TryParse(versionToCheck, out nugetVersion))
-                {
-                    updatedVersions.Add(versionToCheck);
-                }
-            }
-            string packageIndexJsonLocation = feed.GeneratePackageServiceIndex(packageDirectory, updatedVersions);
-            await UploadAsync(CancellationToken, packageIndexJsonLocation, packageDirectory + "/index.json", clientThrottle, true);
-        }
+                Sources = new List<SleetSource>
+                    {
+                       source
+                    }
+            };
 
-        private async Task UploadAsync(CancellationToken ct, string item, string uploadPath, SemaphoreSlim clientThrottle, bool allowOverwrite)
-        {
-            if (!File.Exists(item))
-                throw new Exception(string.Format("The file '{0}' does not exist.", item));
+            LocalSettings settings = new LocalSettings
+            {
+                Json = JObject.FromObject(sleetSettings)
+            };
 
-            await clientThrottle.WaitAsync();
-            string leaseId = string.Empty;
-            //this defines the lease for 15 seconds (max is 60) and 3000 milliseconds between requests
-            AzureBlobLease blobLease = new AzureBlobLease(feed.AccountName, feed.AccountKey, string.Empty, feed.ContainerName, uploadPath, Log, "15", "5000");
+            return settings;
+        }
 
-            bool isLeaseRequired = allowOverwrite && await feed.CheckIfBlobExists(uploadPath);
+        private AzureFileSystem GetAzureFileSystem()
+        {
+            CloudStorageAccount storageAccount = CloudStorageAccount.Parse(source.ConnectionString);
```

Comment:
Can and should this be cached for all the pushes?

Reply:
Talked offline.